main.go

```diff
@@ -472,13 +472,17 @@ func aggregateTimemap(urir string, dttmp *time.Time, sess *Session) (basetm *lis
 }
 
 func parseURI(uri string) (urir string, err error) {
-	if !regs["isprtcl"].MatchString(uri) {
-		uri = "http://" + uri
+	uescd, err := url.QueryUnescape(uri)
```

Comment:
Introducing the new variable `uescd` is unnecessary, which causes more changes down the line than needed. It could have been `uri` instead, because the unescaped URI is not needed anywhere after this.

Reply:
Since the `url.QueryUnescape()` might return a non-`nil` `err`, we need to check for that, otherwise further down it may crash the program.

Reply:
I would recommend using `url.PathUnescape()` here, because that would not convert `+` to ` `. Also, I am not sure why you are getting `%20` for spaces after unescaping, because my tests outside of this codebase are resulting in regular spaces.

Reply:
Maybe, we can replace any regular spaces with `%20` after unescaping, if necessary.

Reply:
I intended to use PathUnescape and not QueryUnescape but was testing out the functionality of the different functions. I will change it.

Reply:
From my reading, if PathUnescape sets err, that err will be readable as having been set when the function returns. That same `err` is also potential set by `url.Parse(uescd)`. In practice, should the same `err` variable be used in scenarios like this if returned or will the former be overridden if not set by the latter (i.e., cleared)?

Reply:
Question here, @ibnesayeed: `uri` is the name of the argument passed in. I know in other languages, one should not overwrite this value, hence the new var `uescd` to store the value after transforming it with `url.PathUnescape()`. Please advise.

Reply:
In Go, it is a common practice to use the guard-clause, that is, check for non-`nil` error each time there is a possibility of returned error and deal with it immediately before going any further down. This was issues can be addressed as soon as identified and then the same variable (traditionally `err`) can be reused. In this particular case, if unescaping returned a non-`nil` error, then the unescaped URL will be an empty string that you will be trying to work on. Further down, if the `url.Parse` returned a `nil` error, then the earlier error will be overwritten and lost. One situation where the unescaping might set an error is if the percent-encoding does not have two hex codes characters (e.g., `z%2z`).

Reply:
Mutating function parameters is a mad idea when they have undesired consequences outside of the function, for example, when a parameter is passed by reference and the state of the passed object will be affected outside the boundaries of the function (sometimes, even that is a desired thing).

Reply:
If the function itself needs to preserve the original form of the parameter for something like caparison, then it is not wise to mutate it, storing any derivatives in an alternate variable would be the way to go in those cases.